pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -360,10 +360,14 @@ func (r *ReconcileSyncSetInstance) syncDeletedSyncSetInstance(ssi *hivev1.SyncSe
 
 	ssiLog.Info("deleting syncset resources on target cluster")
 	err = r.deleteSyncSetResources(ssi, dynamicClient, ssiLog)
-	if err == nil {
-		return reconcile.Result{}, r.removeSyncSetInstanceFinalizer(ssi, ssiLog)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	err = r.deleteSyncSetSecretReferences(ssi, dynamicClient, ssiLog)
+	if err != nil {
+		return reconcile.Result{}, err
```

Comment:
@abutcher Is it a safe assumption that secrets are the only objects that need to get attached to syncsets?

Reply:
Generic resources (objects) and secrets thus far.